### PR TITLE
Make class level fields static where applicable

### PR DESCRIPTION
**Pull request by KoenDG**
This very small PR is related to issue #24226 it is small exactly to keep it readable.

A field may be static if it is declared final, and is initialized with a constant, bringing a small performance gain.

===========

1. Contributor agreement: Has been signed

2. Contributor guidelines: I have read them.

3. Gradle check: build successful.

4. PR is against master

5. The PR is not OS specific.

6. I'm not part of a class.

### CODE REVIEW

**Review by nik9000 — CHANGES_REQUESTED** [review]
I believe all of these are constants. Can you change them to CONSTANT_CASE while you are making them static?

**Comment by KoenDG** [comment]
@nik9000 Done as requested.

File: core/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java [hunk 1]
```diff
@@ -313,11 +313,11 @@ boolean isMemoryLocked() {
     static class MaxNumberOfThreadsCheck implements BootstrapCheck {
 
         // this should be plenty for machines up to 256 cores
-        private final long maxNumberOfThreadsThreshold = 1 << 12;
+        private static final long MAX_NUMBER_THREADS_THRESHOLD = 1 << 12;
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java]: Please add the word "of" back.

**Review by jasontedor — CHANGES_REQUESTED** [review]
I noticed one thing, it was probably a typo.

**Comment by KoenDG** [comment]
@jasontedor fixed.

**Comment by KoenDG** [comment]
Master has been pushed into this branch again. Please advise further actions.

**Pull request merged**